pkg/storaged/lvol-tabs.jsx

```diff
@@ -272,14 +354,60 @@
     if (block && block.IdType == "crypto_LUKS" && block.IdVersion == 2)
         passphrase_fields = existing_passphrase_fields(_("Resizing an encrypted filesystem requires unlocking the disk. Please provide a current disk passphrase."));
 
-    if (!usage.Teardown && size_fields.length + passphrase_fields.length === 0) {
+    if (!has_subvols && !usage.Teardown && size_fields.length + passphrase_fields.length === 0) {
         return lvol_and_fsys_resize(client, lvol, grow_size, info.grow_needs_unmount, null);
     }
 
+    function prepare_pvs(pvs) {
+        if (!pvs)
+            return pvs;
+
+        pvs = pvs.map(spc => spc.block.path);
+
+        if (!has_subvols)
+            return pvs;
+
+        const subvol_pvs = [];
+
+        // Step 1: Find PVs that are already used by a subvolume
+        subvols.forEach((sv, idx) => {
+            subvol_pvs[idx] = null;
+            for (const pv in sv) {
+                if (pvs.indexOf(pv) >= 0 && subvol_pvs.indexOf(pv) == -1) {
+                    subvol_pvs[idx] = pv;
+                    break;
+                }
+            }
+        });
+
+        // Step 2: Use the rest for the leftover subvolumes
+        subvols.forEach((sv, idx) => {
+            if (!subvol_pvs[idx]) {
+                for (const pv of pvs) {
+                    if (subvol_pvs.indexOf(pv) == -1) {
+                        subvol_pvs[idx] = pv;
+                        break;
+                    }
+                }
+            }
+        });
+
+        return subvol_pvs;
+    }
+
     const dlg = dialog_open({
         Title: _("Grow logical volume"),
         Teardown: TeardownMessage(usage),
+        Body: has_subvols && <div><p>{cockpit.format(_("Exactly $0 physical volumes need to be selected, one for each stripe of the logical volume."), subvols.length)}</p><br /></div>,
         Fields: size_fields.concat(passphrase_fields),
+        update: (dlg, vals, trigger) => {
+            if (vals.pvs) {
+                const max = lvol.Size + max_size(vals.pvs);
+                if (vals.size > max)
+                    dlg.set_values({ size: max });
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17226-20230918-111530-a92123e3-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)